QuantLib/ql/pricingengines/capfloor/blackcapfloorengine.cpp

```diff
@@ -56,8 +57,21 @@ namespace QuantLib {
                        const Handle<YieldTermStructure>& discountCurve,
                        const Handle<OptionletVolatilityStructure>& volatility,
                        Real displacement)
-    : discountCurve_(discountCurve), vol_(volatility),
-      displacement_(displacement) {
+    : discountCurve_(discountCurve), vol_(volatility) {
+        QL_REQUIRE(
+            vol_->optionletStripper()->volatilityType() == ShiftedLognormal,
+            "BlackCapFloorEngine should only be used for vol surfaces stripped "
+            "with shifted log normal model. Options were stripped with model "
+                << vol_->optionletStripper()->volatilityType());
+        if (displacement != Null< Real >()) {
```

Comment:
with #295: the OptionletVolatilityStructure already knows which displacement to apply, so we should take it from there via vol_->displacement()